serde_codegen/src/de.rs

```diff
@@ -536,7 +528,20 @@ fn deserialize_item_enum(
         };
         variant_arms.push(arm);
     }
-    variant_arms.extend(ignored_arm.into_iter());
+
+    let match_variant = if variant_arms.is_empty() {
+        // This is an empty enum like `enum Impossible {}` or an enum in which
+        // all variants have `#[serde(skip_deserializing)]`.
+        quote! {
+            visitor.visit_variant::<__Field>().map(|impossible| match impossible {})
```

Comment:
Added in b01c23b5ee3ba96c286b22fb8f958ea6b2576fde. This case is a little different because `visit_variant` returns `Result<__Field>` rather than `Result<Option<__Field>>`.